tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CompleteOrderService.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using NHSD.BuyingCatalogue.EmailClient;
+using NHSD.BuyingCatalogue.Ordering.Api.Services.CompleteOrder;
+using NHSD.BuyingCatalogue.Ordering.Api.Settings;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders.Services;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Application.Services;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NHSD.BuyingCatalogue.Ordering.Domain.Results;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Services
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    internal sealed class CompleteOrderServiceTests
+    {
+        [TestCase(false, true, true, true)]
+        [TestCase(true, false, true, true)]
+        [TestCase(true, true, false, true)]
+        [TestCase(true, true, true, false)]
+        public void Constructor_NullParameter_ThrowsArgumentNullException(
+            bool hasIdentityService,
+            bool hasOrderRepository,
+            bool hasEmailService,
+            bool hasPurchasingSettings)
+        {
+            var builder = CompleteOrderServiceBuilder
+                .Create()
+                .WithIdentityService(hasIdentityService ? Mock.Of<IIdentityService>() : null)
+                .WithOrderRepository(hasOrderRepository ? Mock.Of<IOrderRepository>() : null)
+                .WithEmailService(hasEmailService ? Mock.Of<IEmailService>() : null)
+                .WithPurchasingSettings(hasPurchasingSettings ? Mock.Of<PurchasingSettings>() : null);
+
+            Assert.Throws<ArgumentNullException>(() => builder.Build());
+        }
+
+        [Test]
+        public void CompleteAsync_NullRequest_ThrowsArgumentNullException()
+        {
+            var context = CompleteOrderServiceTestContext.Setup();
+
+            Assert.ThrowsAsync<ArgumentNullException>(() => context.CompleteOrderService.CompleteAsync(null));
+        }
+
+        [Test]
+        public async Task CompleteAsync_CompleteOrderRequest_ReturnsSuccessfulResult()
+        {
+            var context = CompleteOrderServiceTestContext.Setup();
+
+            var actual = await context.CompleteOrderService.CompleteAsync(context.CompleteOrderRequest);
+
+            actual.Should().Be(Result.Success());
+        }
+
+        [Test]
+        public async Task CompleteAsync_IncompleteOrderRequest_ReturnsFailureResult()
+        {
+            var context = CompleteOrderServiceTestContext.Setup();
+
+            var actual = await context.CompleteOrderService.CompleteAsync(
+                new CompleteOrderRequest(OrderBuilder.Create().Build()));
+
+            actual.Should().Be(Result.Failure(new ErrorDetails("OrderNotComplete")));
+        }
+
+        [Test]
+        public async Task CompleteAsync_IdentityService_CalledTwice()
```

Comment:
A test to verify that the Order is being updated with the data received from these calls would be useful